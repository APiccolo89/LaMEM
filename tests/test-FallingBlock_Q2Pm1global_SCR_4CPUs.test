EXEC            = ../bin/LaMEM -ParamFile ./input/FallingBlock_SCR.dat -vpt_element Q2Pm1_global  -AddRandomNoiseParticles 0 -time_end 1 -nel_x 4 -nel_y 4 -nel_z 8
NPROCS          = 4
TEST_TYPE       = tolerance_diff_pl,1.0e-4
