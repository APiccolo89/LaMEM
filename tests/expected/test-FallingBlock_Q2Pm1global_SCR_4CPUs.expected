# -------------------------------------------------------------------------- 
#                   Lithosphere and Mantle Evolution Model                   
#              Current Revision: 4812 - 2013-10-01 23:57:18		     
#  Modified items: LaMEM_Particles.c, Elements.c, Assembly_FDSTAG.c, fdstag/JacRes.c, 
#     Compiled: Date: Oct  9 2013 - Time: 23:02:54 - Mode:  	    
# -------------------------------------------------------------------------- 
 Parsing input file : ./input/FallingBlock_SCR.dat 
# Adding PetscOption: -vpt_element Q1P0 
# Adding PetscOption: -restart 0 
# Adding PetscOption: -save_breakpoints 0 
# Adding PetscOption: -scr_ksp_rtol 1.0e-4 
# Adding PetscOption: -scr_ksp_atol 1.0e-13 
# Adding PetscOption: -scr_pc_type bjacobi 
# Adding PetscOption: -vs_ksp_type fgmres 
# Adding PetscOption: -vs_pc_type fieldsplit 
# Adding PetscOption: -vs_fieldsplit_ksp_type fgmres 
# Adding PetscOption: -vs_fieldsplit_ksp_rtol 1.0e-2 
# Adding PetscOption: -vs_pc_fieldsplit_block_size 3 
# Adding PetscOption: -vs_pc_fieldsplit_type SYMMETRIC_MULTIPLICATIVE 
# LithColDepth[0] = 0 
# LithColDens[0] = 0 
# LithColDens[1] = 0 
# --------------------------------------------------------------------------
# Phase material parameters read from ./input/FallingBlock_SCR.dat: 
#    phase_1 (id=1), VISCOSITY, constant [0,0,0] = 1.000000e+03 
#    phase_1 (id=1), ELASTICITY, constant [0,1,0] = 5.000000e+10, 1.000000e+00 
#    phase_1 (id=1), DENSITY, temperature-dependent [0,2,0] = 2.000000e+00, 0.000000e+00, 0.000000e+00 
#    phase_1 (id=1), ENERGY, constant [0,3,0] = 3.000000e+00 1.050000e+03 0.000000e+00 0.000000e+00 
#    phase_0 (id=0), VISCOSITY, constant [1,0,0] = 1.000000e+00 
#    phase_0 (id=0), ELASTICITY, constant [1,1,0] = 5.000000e+10, 1.000000e+00 
#    phase_0 (id=0), DENSITY, temperature-dependent [1,2,0] = 1.000000e+00, 0.000000e+00, 0.000000e+00 
#    phase_0 (id=0), ENERGY, constant [1,3,0] = 3.000000e+00 1.050000e+03 0.000000e+00 0.000000e+00 
 Stokes solver             : Schur Complement Reduction
 Velocity solver           : User-defined (set options with prefix vs_)
 Note: Galerkin geometric multigrid is deactivated for user-defined solver 
 Element-type used         : Q2Pm1_global;  [Q1P0 Q2Pm1_global Q2Pm1_local Q1Q1 FDSTAG] (change with -vpt_element)
 Total # of cpu's          : 4 
 Total # of velocity dof's : 4131 
 Computing mean material per element using arithmetic averaging from integration points 
 Employing code in a Lagrangian manner 
 Gravity angle with z-axis =  0 
 Approximate Aspect ratio of cells [dx/dz, dy/dz] = [2 2] 
 WARNING: At least 3 particles required in x-direction for Q2 elements; I have increased this 
 WARNING: At least 3 particles required in y-direction for Q2 elements; I have increased this 
 WARNING: At least 3 particles required in z-direction for Q2 elements; I have increased this 
 Number of tracers/cell in [x,y,z]-direction = [3,3,3] 
 NumSurfaceNodes=5  nz=17 
 # Not employing nonlinear iterations 
 BC employed: BC.[Left Right; Front Back; Lower Upper]=[1 1; 1 1; 1 2] 
# Initial time step: 0.001 years 
 Input is assumed to be in non-dimensional units 
 Fine Grid parameters, nel=[4,4,8], cpu=[1,1,4] 
# Finished initializing Stokes matrices : 0.046129 s 
# Not adding random noise to initial particle distribution. 
Initialized particles 
#rank=0, Attributing 864 particles to elements took 0.0195529 s 
The total # of particles present is 3456 
 *** Using a falling block setup *** 
 Falling block coordinates:  [left,right]=[0.250000,0.750000]; [front,back]=[0.250000,0.750000]; [bottom,top]=[0.250000,0.750000] 
Saved initial particle distribution to directory ./InitialParticles 
The total memory used by integration points is 9.123840e-01 MB
The max memory used by integration points is 2.280960e-01 MB
The min memory used by integration points is 2.280960e-01 MB
#  Interpolating meshes and temperatures took 0.000334978 s 
#  Forming FEM stiffness matrix mx,my,mz=(9, 9, 17) ...   finished. [0.0568061 s]
 minDiagonalRatio=1 [Critical = 0.55; change with -CriticalDiagonalRatio]  
#  Forming stiffness matrixes took 0.0579801 s 
SCR: Viscosity structure info: min(0)=1.000000e+00 : max(148)=1.000000e+03 : contrast=1.000000e+03 
  Residual norms for scr_ solve.
  0 KSP Residual norm 2.041878399278e-02 
  1 KSP Residual norm 7.306492121201e-04 
  2 KSP Residual norm 1.173929784768e-04 
  3 KSP Residual norm 8.201023422232e-05 
  4 KSP Residual norm 1.083849612833e-05 
  5 KSP Residual norm 6.302921182962e-06 
  6 KSP Residual norm 2.067408086357e-06 
  7 KSP Residual norm 9.811390445976e-07 
Linear solve converged due to CONVERGED_RTOL iterations 7
# Time: p solve = 4.29e-01 
# Time: u solve = 9.84e-02 
#  Finished Stokes solve after 0.686522 s 
------------------------------------------
Stokes solver summary: 
  divergence: 
    Div_min = 3.218233266166e-11 
    Div_max = 2.316887578796e-07 
    |Div|_2 = 1.010849073298e-06 
    |Div|_1 = 1.462594148802e-05 
  momentum: 
    mRes_min = 0.000000000000e+00 
    mRes_max = 3.315714932055e-08 
    |mRes|_2 = 1.852431744439e-07 
    |mRes|_1 = 5.181822958142e-06 
  velocity: 
    u_min   = -2.961467373432e-03 
    u_max   = 3.322906441662e-03 
    |u|_2   = 7.414456377850e-02 
    |u|_1   = 2.953126494903e+00 
  pressure: 
    p_min   = -1.713642008949e+00 
    p_max   = 1.221225294649e+00 
    |p|_2   = 1.445118567309e+01 
    |p|_1   = 2.366292475834e+02 
------------------------------------------
The total memory used by integration points is 9.123840e-01 MB
The max memory used by integration points is 2.280960e-01 MB
The min memory used by integration points is 2.280960e-01 MB
#  Solving for velocity took 0.750268 s 
#  Forming temperature stiffness matrix mx,my,mz=(9, 9, 17) ...  finished. 
#  Forming temperature stiffness matrixes took 0.00630593 s 
#  Solving for temperature took 0.0011251 s 
The total memory used by integration points is 9.123840e-01 MB
The max memory used by integration points is 2.280960e-01 MB
The min memory used by integration points is 2.280960e-01 MB
The total memory used by particles is 9.676800e-01 MB
The max memory used by particles is 2.419200e-01 MB
The min memory used by particles is 2.419200e-01 MB
#  Updating properties @ integration points and particles took 0.00491285 s 
AdvectionMethod = 1 
  Starting particle advection 
  Finished particle advection 
Minimum z-level phase 1 0.270832 
# Writing output to existing directory Timestep_000000 
# Saved MATLAB output in file Timestep_000000/FB_test.0.1000000.out 
# Saved Paraview VTS Velocity output in file FB_test_velocity_step000000-mesh.pvts 
# Saved Paraview VTS Temperature output in file FB_test_temperature_step000000-mesh.pvts 
# Saved Paraview VTS Surface Topography output in file FB_test_Surface_topography_step000000-mesh.pvts 
# Saved Paraview VTS Bottom Topography output in file FB_test_Bottom_topography_step000000-mesh.pvts 
# Saved Paraview VTS point-wise quadrature output in file FB_test_quadrature_step000000-mesh.pvts 
 Maximum velocity  0.00332291    
 Time = 0.001, dt=1e-05   
# Finished timestep 0 out of 1 in 0.776213 s
  
# Total time required: 0.857962 s 
#PETSc Option Table entries:
-AddRandomNoiseParticles 0
-ParamFile ./input/FallingBlock_SCR.dat
-nel_x 4
-nel_y 4
-nel_z 8
-options_left
-restart 0
-save_breakpoints 0
-scr_ksp_atol 1.0e-13
-scr_ksp_converged_reason
-scr_ksp_max_it 1000
-scr_ksp_monitor
-scr_ksp_rtol 1.0e-4
-scr_ksp_type fgmres
-scr_pc_type bjacobi
-time_end 1
-vpt_element Q2Pm1_global
-vs_fieldsplit_ksp_rtol 1.0e-2
-vs_fieldsplit_ksp_type fgmres
-vs_ksp_type fgmres
-vs_pc_fieldsplit_block_size 3
-vs_pc_fieldsplit_type SYMMETRIC_MULTIPLICATIVE
-vs_pc_type fieldsplit
#End of PETSc Option Table entries
There are no unused options.
